src/back/GameLauncher.ts

```diff
@@ -394,13 +394,30 @@ function escapeWin(str: string): string {
  * ( According to this: https://stackoverflow.com/questions/15783701/which-characters-need-to-be-escaped-when-using-bash )
  */
 function escapeLinuxArgs(str: string): string {
-  return (
-    splitQuotes(str)
-    .reduce((acc, val, i) => acc + ((i % 2 === 0)
-      ? val.replace(/[~`#$&*()\\|[\]{};<>?!]/g, '\\$&')
-      : '"' + val.replace(/[$!\\]/g, '\\$&') + '"'
-    ), '')
-  );
+  // Characters to always escape:
+  const escapeChars: string[] = ['~','`','#','$','&','*','(',')','\\\\','|','[','\\]','{','}',';','<','>','?','!'];
+  if (str.match(/\'/gi) == null || (str.match(/\'/gi)!.join('').length) % 2 == 0) {
```

Comment:
You're getting a non-null assertion because you're comparing 2 (potentionally) different results. Set it to a variable first.

```
  const match = str.match(/'/gi);
  if (match == null || match.join('').length % 2 == 0) {
  ...
```

The escapes inside your regexes should be fine to remove, both /'/gi and /\'/gi will match the same.

Reply:
Understood on the non-null assertion, I've added the variable.

For the escapes, I've confirmed that if I remove even one of the two backslashes currently before the closing square bracket, affected games will not start. It needs two backslashes in the way it's currently written. There aren't actually any games in the database that have a backslash in their launch command, but testing here, it looks like double-escaping is necessary for the backslash-escape as well: [TS Playground](https://www.typescriptlang.org/play?#code/DYUwLgBCDODGCGAHEBhAFvATtAXBaYmAlgHYDmA2gLoQC8EFA5AH6MA0jABu4wMQ8ASHgDIeAKh4AKHgEoeAHXk8APjyYd5VHgG8eAXx4BuHgB4eAPh4B+HgEJ2ERksZVDAKFgB7EtE+gAdMCeZJIwCMjoWND+AFaepJIARIkyMu4AblgQALYAngBqWETwAEagdI4A4vDZIErQEF7Z2fAkACYQZJ4wEIoQaCCYIFT+jB7evgFBIXmFxKWgMuM+fiCBwZKzRQtrQ4jA8LAgkiQgAO4QAEogZACiAB6ISRSJEADUUHBIqBjYsfEkJIpd4QRJURJsRxkRgySFOeQCUSpNzLSZrabSGHuNxhb6RbAVJisDjcDj8DhCDiiDgSDjSDhyDSKJQcVQcdTwrQcXQcAwcYwcMwcSwcGwcexw5yuVGrdYhXERX7ROIJZKpDJZLbzMogCqMaq1eqNTzNVodLo9PoDIYjMZeFZTDZa4o6pb2tFyzYFbY6-x7A5HE7nK43B5PRIvEEKn5Rf6q4EfMEQqEwyUIpEyIA)

I think what's happening is that the first escape is "used" when creating the array, and the second one is "used" when creating the RegExp object (where it's actually needed.) If you'd prefer, I could just put the "hard-coded" string back into the `val.replace` function (with a `'` in the second case and no `'` in the first case.) I just thought it would be more elegant to not duplicate that list of characters, since it's only a single-character difference. (I just now remembered that part of the point was to only need the `reduce` function there once, so I've gotten rid of the duplicate to further clean it up in the latest commit.)